### PR TITLE
Fix ErrorResponse interface to satisfy TF SDK

### LINKED ISSUE #195: Plugin panics on terraform destroy after failed terraform apply
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

```
Terraform v1.0.8
on linux_amd64
+ provider registry.terraform.io/equinix/metal v3.2.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/poseidon/ct v0.6.1
```

### Affected Resource(s)

* metal_device

### Terraform Configuration Files

https://github.com/arianvp/packet-ipv6-kubernetes/blob/metal/provision.tf

### Debug Output

https://gist.github.com/arianvp/6c670ca5be1f4541c1c12d9b67942409

### Panic Output

https://gist.github.com/arianvp/6c670ca5be1f4541c1c12d9b67942409

### Expected Behavior

Resources get destroyed

### Actual Behavior

Terraform panics

### Steps to Reproduce

1. `terraform apply`
2. get a:
 ```
metal_device.worker[1]: Creating...
metal_device.worker[0]: Creating...
╷
│ Error: Sorry! You have reached the maximum number of t1.small.x86 devices for this project (5). The Equinix Metal team has been notified of this and will be in touch shortly. If this is urgent, please email support@equinixmetal.com to request additional deployment capacity.
│ 
│ 
│   with metal_device.worker[0],
│   on provision.tf line 111, in resource "metal_device" "worker":
│  111: resource "metal_device" "worker" {
│ 
╵
╷
│ Error: Sorry! You have reached the maximum number of t1.small.x86 devices for this project (5). The Equinix Metal team has been notified of this and will be in touch shortly. If this is urgent, please email support@equinixmetal.com to request additional deployment capacity.
│ 
│ 
│   with metal_device.worker[1],
│   on provision.tf line 111, in resource "metal_device" "worker":
│  111: resource "metal_device" "worker" {

```
2. Ran out of capacity of `t1.small` instances
3. Decided to run  `terraform destroy` to destroy the entire project
4. Terraform panics

### References

* #0000

### PR DESCRIPTION
**Pull request by t0mk**
This PR fixes the situation where the EM API returns HTTP 422, and `errors: []` in the body. Like this:

```
---[ REQUEST ]---------------------------------------
DELETE /metal/v1/projects/0a286170-17fc-43a0-9b68-4ddfb41c4f4c HTTP/1.1
Host: api.equinix.com
User-Agent: HashiCorp Terraform/0.11+compatible (+https://www.terraform.io) Terraform Plugin SDK/2.7.0 terraform-provider-metal/dev packngo/v0.19.1
Connection: close
Accept: application/json
Content-Type: application/json
X-Consumer-Token: aZ9GmqHTPtxevvFq9SK3Pi2yr9YCbRzduCSXF2SNem5sjB91mDq7Th3ZwTtRqMWZ
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2021-10-18T18:19:22.374+0300
2021-10-18T18:19:23.206+0300 [INFO]  provider.terraform-provider-metal: 2021/10/18 18:19:23 [DEBUG] Equinix Metal API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 422 Unprocessable Entity
Connection: close
Content-Length: 13
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Date: Mon, 18 Oct 2021 15:19:23 GMT
Strict-Transport-Security: max-age=15724800; includeSubDomains
X-Request-Id: 6e1b741e5249ce3f03812ebad8fd1770

{
 "errors": []
}
```

Hopefully fixes #195

### CODE REVIEW

File: metal/errors.go [hunk 1]
```diff
@@ -73,6 +75,18 @@ type ErrorResponse struct {
 	IsAPIError bool
 }
 
+func (er ErrorResponse) Error() string {
```
Comment (displague) [inline metal/errors.go:78]: I think this should be a pointer type.

I'm not sure what to expect from this function in practice, especially with different combinations of status codes, IsAPIError flags, and the original errors.

Comment (displague) [reply metal/errors.go:78]: A table test of this type that demonstrates some of the expected and possible values would help.

Comment (t0mk) [reply metal/errors.go:78]: @displague The panic that this is fixing happens somewhere deep in the TF SDK code. Somehow we pass the ErrorResponse struct to the SDK as an error interface, and when it's about be cast, the code panics because the ErrorResponse doesn't have the Erorr() method. If I recall correctly.

To invoke this error, I had to work with a Metal support eng to set up limit for one of my projects, so that I can hit the situation from #195. When I managed to invoke it and debug the error, it took about a week to remove the project limit (or some other restriction). Right now I can't test it, and I don't know if this will work if we do `func (er *ErrorResponse) Error() string {`. 

As the panic was popping up in TF SDK code, to properly test this we'd need to build the erring response and pass it to initialized provider. That would mean to create a mock API endpoint, and (since the error that this is fixing happens on Delete) mock responses for device creation POST, device resource GET, and finally the DELETE. I wouldn't know where to start with this.

Another way to mitigate #195 without merging this is to fix the API behavior, so that it never sends `errors: []` in a response.

TLDR, I can't verify it this will work if I use pointer type, and I can't imagine a doable way to test this.

**Comment by t0mk** [comment]
@displague Looks good, and seem to work. I think we can't run this withou TF_ACC=1 because `resource.Test` is checking it inside. I didn't find any other provider which would be using mock to this extent.

Although, if I would be about to put effort into fixing the error mess, I would rather remove the ErrorResponse type from packngo, and treat body of HTTP error responses only as string. I would also remove the ErrorResponse struct here from the provider. That would clean the code a bit, because there wouldn't be casting of ErrorResponse, packngo.ErrorResponse and http.Response. Once the code would be cleaner, it would be easier to fix issues like this.

File: metal/errors.go (hunk 1 above)
Comment (displague) [reply metal/errors.go]: (as we discussed in a DM) 40bf28c7be0db195afb56b849d09f4f5263f6003 adds mocks to simplify testing of the error conditions.
